code/controllers/subsystems/pathfinder.dm

```diff
@@ -0,0 +1,47 @@
+SUBSYSTEM_DEF(pathfinder)
+	name = "Pathfinder"
+	init_order = SS_INIT_PATH
+	flags = SS_NO_FIRE
+
+	var/datum/flowcache/mobs
+	var/static/space_type_cache
+
+/datum/controller/subsystem/pathfinder/Initialize()
+	space_type_cache = typecacheof(/turf/space)
+	mobs = new(10)
+	return ..()
+
+/datum/flowcache
+	var/lcount
+	var/run
+	var/free
+	var/list/flow
+
+/datum/flowcache/New(n)
+	. = ..()
+	lcount = n
+	run = 0
+	free = 1
+	flow = new /list(lcount)
+
+/datum/flowcache/proc/getfree(atom/M)
```

Comment:
This is a bit over my head, what is `getfree` doing? Some more informative names would be good.